src/main/kotlin/com/atlassian/performance/tools/virtualusers/api/users/RestUserGenerator.kt

```diff
@@ -23,14 +27,14 @@ class RestUserGenerator(readTimeout: Duration) : UserGenerator {
         options: VirtualUserOptions
     ): User {
         val target = options.target
-        val uuid = UUID.randomUUID()
-        val userName = "jpt-$uuid"
+        val displayName = nameGenerator.pickRandomUnique()
```

Comment:
How are the name collisions handled?

I see two cases here:
1. The instance already contains the user (from the dataset or previous test run).
2. Another VU tries to generate the same user (MulticastVUs).

BTW: 
1. The pool can be depleted, so it limits the number of VUs we can generate. 
2. The generator may randomly fail to deliver the name. See the implementation:
```
 fun pickRandomUnique(): String {
        for (i in 1..10) {
            val picked = pickRandom()
            if (namesGeneratedSoFar.add(picked)) {
                return picked
            }
        }
        throw IllegalStateException("Random name pool is depleted")
    }
```

Reply:
The old version of the generator can reach about 90 000 names before the depletion starts. The new version will switch to generating two given names once collisions start, which means we'll get more names than we'll ever need (millions).

But obviously the 2 first items you brought up are way more important, we need retries on these REST calls.